clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1214,7 +1214,8 @@ void KernelObjVisitor::visitRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
                                    const CXXRecordDecl *Wrapper,
                                    QualType RecordTy,
                                    HandlerTys &... Handlers) {
-  if (RecordTy->getAsRecordDecl()->hasAttr<SYCLRequiresDecompositionAttr>()) {
+  if (RecordTy->getAsRecordDecl() != nullptr &&
+      RecordTy->getAsRecordDecl()->hasAttr<SYCLRequiresDecompositionAttr>()) {
```

Comment:
I suggest calling `getAsRecordDecl()` once.
```suggestion
  RecordDecl *RD = RecordTy->getAsRecordDecl();
  if (RD && RD->hasAttr<SYCLRequiresDecompositionAttr>()) {
```